src/server/test/web/readingsCompareMeterQuantity.js

```diff
@@ -74,7 +74,95 @@ mocha.describe('readings API', () => {
 
 				// Add C6 here
 
-				// Add C8 here
+				mocha.it('C8: 1 day shift end 2022-10-31 17:00:00 for 15 minute reading intervals and quantity units & kWh as MJ', async () => {
+					// adding units u1, u2, u3
+					const unitData = [
```

Comment:
This way of adding units, conversions & meter works fine. However, you might want to look at ``src/server/test/web/readingsBarGroupQuantity.js`` in test case BG8 that uses the same units but gets the standard testing ones rather than redefining.

Reply:
Hello Huss, 

Thank you for the reply and the descriptive comment you provided. 
Is this how you want us to modify our code based on the feedback provided? 

```
mocha.it('C8: 1 day shift end 2022-10-31 17:00:00 for 15 minute reading intervals and quantity units & kWh as MJ', async () => {
    // Use predefined unit and conversion data
    const unitData = unitDatakWh.concat([
        {
            name: 'MJ',
            identifier: 'megaJoules',
            unitRepresent: Unit.unitRepresentType.QUANTITY,
            secInRate: 3600,
            typeOfUnit: Unit.unitType.UNIT,
            suffix: '',
            displayable: Unit.displayableType.ALL,
            preferredDisplay: false,
            note: 'MJ'
        }
    ]);
    const conversionData = conversionDatakWh.concat([
        {
            sourceName: 'kWh',
            destinationName: 'MJ',
            bidirectional: true,
            slope: 3.6,
            intercept: 0,
            note: 'kWh → MJ'
        }
    ]);
    // Prepare test with the standard data
    await prepareTest(unitData, conversionData, meterDatakWhGroups, groupDatakWh);
    // Get the unit ID
    const unitId = await getUnitId('MJ');
    const expected = await parseExpectedCsv('src/server/test/web/readingsData/expected_bar_group_ri_15-20_mu_kWh_gu_MJ_st_-inf_et_inf_bd_1.csv');
    // Make the API call
    const res = await chai.request(app).get(`/api/compareReadings/meters/${METER_ID}`)
        .query({
            curr_start: '2022-10-31 00:00:00',
            curr_end: '2022-10-31 17:00:00',
            shift: 'P1D',
            graphicUnitId: unitId
        });
    // Compare the results with the expected values
    expectCompareToEqualExpected(res, expected);
});
```

Reply:
Yes, that seems correct and what I was trying to say.

Reply:
Hello Huss, 

We have implemented your suggestions into our code.